web/client/codechecker_client/cmd/store.py

```diff
@@ -455,12 +455,29 @@ def filter_source_files_with_comments(source_file_info, main_report_positions):
 
         jobs.append((file_path, lines))
 
-    with concurrent.futures.ProcessPoolExecutor() as executor:
-        for job, comments in zip(jobs,
-                                 executor.map(scan_for_review_comment, jobs)):
-            file_path, _ = job
-            if comments:
-                files_with_comment.add(file_path)
+    def _populate_files(job, comments, file_list):
+        """updates file_listlist with files & comments"""
+        file_path, _ = job
+        if comments:
+            file_list.add(file_path)
+        return file_list
+
+    # Currently ProcessPoolExecutor fails completely in windows
+    # Reason is most likely combination of venv and fork() not
+    # being present in windows, so stuff like setting up
+    # PYTHONPATH in parent CodeChecker before store is executed
+    # are lost. Maybe
+    if util.is_windows():
+        for job, comments in zip(jobs, map(scan_for_review_comment, jobs)):
+            files_with_comment = \
+                    _populate_files(job, comments, files_with_comment)
```

Comment:
We are still duplication a couple of thing because of the `map vs executor.map` functions. Also we don't extend the set here, we reassign a new value in every iteration.
What do you think about the following approach:
```py
def get_source_file_with_comments(jobs, zip_iter=map):
    """
    Get source files where there is any codechecker review comment at the main
    report positions.
    """
    files_with_comment = set()

    for job, comments in zip(jobs,
                             zip_iter(scan_for_review_comment, jobs)):
        file_path, _ = job
        if comments:
            files_with_comment.add(file_path)

    return files_with_comment


def filter_source_files_with_comments(source_file_info, main_report_positions):
    """Collect the source files where there is any codechecker review
    comment at the main report positions.
    """
    jobs = []
    for file_path, v in source_file_info.items():
        if not bool(v):
            # missing file
            continue
        lines = [rep.line for rep in main_report_positions
                 if rep.filepath == file_path]

        jobs.append((file_path, lines))

    # Currently ProcessPoolExecutor fails completely in windows.
    # Reason is most likely combination of venv and fork() not
    # being present in windows, so stuff like setting up
    # PYTHONPATH in parent CodeChecker before store is executed
    # are lost.
    if sys.platform == "win32":
        return get_source_file_with_comments(jobs)
    else:
        with concurrent.futures.ProcessPoolExecutor() as executor:
            return get_source_file_with_comments(jobs, executor.map)
```